src/main/java/org/spongepowered/api/world/explosion/ResistanceCalculator.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.world.explosion;
+
+import com.flowpowered.math.vector.Vector3i;
+import org.spongepowered.api.block.BlockState;
+
+/**
+ * A Functional Interface for the fetching of a Block's explosion resistance during an explosion
+ */
+@FunctionalInterface
+public interface ResistanceCalculator {
+    /**
+     * A blueprint for a lambda that takes three variables; BlockState, Vector3i, and Explosion and returns a float.
```

Comment:
The javadoc comment is pointless because we already know what has been written here. An interface is a blueprint, and we can see return and parameter types.

Focus on what the method does and the context around that in a doc comment - it should tell you what the code cannot. For example, what values can I return? Is it 0 to 100? 0 to 1? Is a negative value valid?

```
Sets the explosion resistance for a block at a given location during an {@link Explosion}.

<p>The resistance must a value between zero and one. A resistance of zero will ensure that the given block will be affected by the explosion. A resistance of one ensures that the block will be unaffected.</p>
```